docs/nextgen-toolkit/cossdev-guide/prepnamespace.rst

```diff
@@ -2,11 +2,72 @@
 
 .. _cossdev-guide-prepnamespace:
 
-Prepare |coss| Namespace
-========================
+Prepare |coss| Namespace and Staging
+====================================
 
-..  note::  This section is still a work-in-progress
+Before paring-away desired |coss| functionality into |uobjs| and |uobjcolls|, we 
+need to prepare the |coss| namespace and switch to a relevant |uberspark| staging.
 
-- prep a folder called helouobjcoll (ref uobjcoll)
-- within that folder add another directory called hello uobj (fref uobj)
-- switch staging to match |coss| architecture and CPU requirements
+We use our running example of ``hello-mul`` |coss| application with sources residing
+within the ``coss-examples/hello-mul/coss-src`` folder.
+
+As a first step, create a top-level folder to house the ``hello-mul`` |uobjcoll|.
+We will use the folder ``uobjcoll`` within the existing ``hello-mul`` source root. i.e., the
+new folder where we will house the refactored ``hello-mul`` |uobjcoll| will be:
+
+``coss-examples/hello-mul/coss-src/uobjcoll``
+
+Within the aforementioned folder, let us create another folder called ``main`` to house
+our only |uobj| within the new ``hello-mul`` |uobjcoll|. i.e., the
+new folder where we will house the the ``main`` |uobj| for the ``hello-mul`` |uobjcoll| will be:
+
+``coss-examples/hello-mul/coss-src/uobjcoll/main``
+
+..  note::  In our example, we choose to create a single |uobj|. In principle, one can
+            create multiple |uobjs| depending on |coss| code-base modules and functionality
+            that is being pared-away into into a |uobjcoll|
+
+At this point, you are ready to create a relevant staging for the ``hello-ml`` |uobjcoll|. 
+|uberspark| staging environments are essentially development workspaces that target a given hardware platform
+with associated CPU(s) and devices. Staging environments also provide configurable settings for 
+the build tools related to the |uobjcoll| development life-cycle such as compilers, assemblers, linkers and 
+verification tools.
+

```

Comment:
I would add note here about the assumed build environment (WSL/Ubuntu), and requirement to change `--root-dir` based upon build environment.  These comments are mentioned at the end, but it might help to note that here.